plonky2/src/hash/keccak.rs

```diff
@@ -49,21 +87,25 @@ impl<F: RichField> PlonkyPermutation<F, KeccakHashConfig> for KeccakPermutation
             .filter(|&word| word < F::ORDER)
             .map(F::from_canonical_u64);
 
-        hash_onion_elems
+        self.state = hash_onion_elems
             .take(SPONGE_WIDTH)
             .collect_vec()
             .try_into()
-            .unwrap()
+            .unwrap();
+    }
+
+    fn squeeze(&self) -> &[F] {
+        &self.state[..Self::RATE]
     }
```

Comment:
Hm, not sure I follow you. `HASH_SIZE` measures the size of the hash state in *bytes*, so it's definitely not the right value to define the range on L98, which is a range of `Field` elements. In fact, based on the other usages in the code, I think `HASH_SIZE` is probably redundant, or at least doesn't need to be in the `Hasher` trait, and I'm going to look into refactoring or removing it.

Maybe that method shouldn't be called `squeeze()`, but I'm pretty certain it should return access to `self.state[..Self::RATE]`, since it is precisely a refactoring of all the places in the code that read `RATE` elements from the permutation (example usage is [here](https://github.com/mir-protocol/plonky2/blob/8d367eedc3018562db55c6b70262caab7901f733/plonky2/src/hash/hashing.rs#L48), and note that the comment there referring to "squeezing" predates this PR). Please feel free to suggest a better name if `squeeze()` though, if you have one!

Reply:
All good; finding appropriate names for things is very important!